site2/website/contributing.md

```diff
@@ -0,0 +1,250 @@
+
+###### Initial setup
+
+1. Import the Pulsar projects
+
+	File
+	-> Import...
+	-> Existing Maven Projects
+	-> Browse to the directory you cloned into and select "pulsar"
+	-> make sure all pulsar projects are selected
+	-> Finalize
+
+You now should have all the Pulsar projects imported into eclipse and should see no compile errors.
+
+### Create a branch in your fork
+
+You’ll work on your contribution in a branch in your own (forked) repository. Create a local branch, initialized with the state of the branch you expect your changes to be merged into. Keep in mind that we use several branches, including `master`, feature-specific, and release-specific branches. If you are unsure, initialize with the state of the `master` branch.
+
+	$ git fetch apache
+	$ git checkout -b <my-branch> apache/master
+
+At this point, you can start making and committing changes to this branch in a standard way.
+
+### Syncing and pushing your branch
+
+Periodically while you work, and certainly before submitting a pull request, you should update your branch with the most recent changes to the target branch.
+
+    $ git pull --rebase
+
+Remember to always use `--rebase` parameter to avoid extraneous merge commits.
+
+Then you can push your local, committed changes to your (forked) repository on GitHub. Since rebase may change that branch's history, you may need to force push. You'll run:
+
+	$ git push <GitHub_user> <my-branch> --force
+
+### Testing
+
+All code should have appropriate unit testing coverage. New code should have new tests in the same contribution. Bug fixes should include a regression test to prevent the issue from reoccurring.
+
+## Review
+
+Once the initial code is complete and the tests pass, it’s time to start the code review process. We review and discuss all code, no matter who authors it. It’s a great way to build community, since you can learn from other developers, and they become familiar with your contribution. It also builds a strong project by encouraging a high quality bar and keeping code consistent throughout the project.
+
+### Create a pull request
+
+Organize your commits to make a committer’s job easier when reviewing. Committers normally prefer multiple small pull requests, instead of a single large pull request. Within a pull request, a relatively small number of commits that break the problem into logical steps is preferred. For most pull requests, you'll squash your changes down to 1 commit. You can use the following command to re-order, squash, edit, or change description of individual commits.
+
+    $ git rebase -i apache/master
+
+You'll then push to your branch on GitHub. Note: when updating your commit after pull request feedback and use squash to get back to one commit, you will need to do a force submit to the branch on your repo.
+
+Navigate to the [Pulsar GitHub Repo](https://github.com/apache/incubator-pulsar) to create a pull request. The title of the pull request should be strictly in the following format:
```

Comment:
> Simple changes, like fixing typos, do not require an associated issue.

v.s.

> The title of the pull request should be strictly in the following format:
 	Issue <Github-issue-#> <Title of the pull request>

Perhaps it'd be better to add "if/unless ...".

Reply:
Good point. Fixed